app/javascript/controllers/viral/sortable_lists/two_lists_selection_controller.js

```diff
@@ -20,33 +17,51 @@ export default class extends Controller {
   connect() {
     this.availableList = document.getElementById(this.availableListValue)
     this.selectedList = document.getElementById(this.selectedListValue)
-    this.fullListItems = this.availableList.querySelectorAll("li")
+    this.allListItems = this.#constructAllListItems(this.availableList, this.selectedList)
+    this.#setInitialSelectAllState(this.availableList, this.addAllTarget)
+    this.#setInitialSelectAllState(this.selectedList, this.removeAllTarget)
     this.selectedList.addEventListener("mouseover", () => { this.#checkButtonStates() })
```

Comment:
**nitpick:** Just noticed this.  Change this in the connect and disconnect methods:
```javascript
this.selectedList.addEventListener("mouseover", () => { this.#checkButtonStates() })
```
to
```javascript
this.selectedList.addEventListener("mouseover", this.#checkButtonStates);
```
The remove actually won't work properly the way you have it.